### ISSUE TITLE
Time Series panel missing points in line style with sparse data

### ISSUE THREAD

**Issue by justinsmalley** [grafana/grafana#48835]
![image](https://user-images.githubusercontent.com/99200643/167187475-0e4e6cc5-3bac-4e4b-a888-17c9e8ad1321.png)
![image](https://user-images.githubusercontent.com/99200643/167187331-44a3e319-087b-4a8d-86e1-491cc7ee4090.png)
![image](https://user-images.githubusercontent.com/99200643/167188055-12a77707-d20f-4cf1-b4fb-af9293a95882.png)

![image](https://user-images.githubusercontent.com/99200643/167187516-cd52e9e3-0a96-43ac-b236-5f4839c2341b.png)
![image](https://user-images.githubusercontent.com/99200643/167187941-6198b53a-1ddb-495f-ad5d-4410b8003424.png)

**What happened**: Using a time series panel in line mode, not all points are displayed. The data is somewhat sparse, and in this case the interval is 2s. There are very few (or zero) lines at this zoom level, however once I zoom out and the interval increases then the lines start appearing. 

When these points are missing, you can mouse over the location and the point will appear. 

**What you expected to happen**:
All data points shown as part of a line OR an individual point.

I *could* turn on Show Points = Always, but then when I zoom out and the line becomes visible, 1000's of individual points start to look messy. Basically, I expect when Style = Line and Show Points = Auto, if a point is part of a line hide the point, or if it's "by itself", the point should be shown.

**How to reproduce it (as minimally and precisely as possible)**:
Create a set of sparse data which is mostly null, but has data randomly spaced 3-30 seconds apart. Zoom such that interval = 1 or 2. Switch back and forth between Line and Point style to see missing data. Also try changing line interpolation mode and it will change which tags are visible and which are not.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.5.0
- Data source type & version: influxdb 1.8
- OS Grafana is installed on: x86-64
- User OS & Browser: MacOS & Chrome
- Grafana plugins:
- Others:

**Comment by leeoniya** [comment]
hmm, i'm not having much luck reproducing this.

![image](https://user-images.githubusercontent.com/43234/167228062-2f7a70cd-50c2-4239-a6bd-6384720526d5.png)


are you able to find a dataset that exhibits the issue when adding a test panel to https://play.grafana.org/d/000000012/grafana-play-home?orgId=1 ?

if you choose `gdev-testdata` for the Data source and `CSV Metric Values` for Scenario, you can create nulls in the data by using sequential commas to produce a similar sparse dataset:

```
1,,,,,,,1,,,,,,,,,,,,,9,,,,,,,,,,,,,,0,,,,,,,,,,,,1,,,,,,,,,,1,,,,,,,,,,,,,,,,,,,,,,1,,,,,,,,1,,,,,,,,1,1,,,,,1,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
```

you can also try using the `CSV Content` Scenario to control both x and y:

```
time,value
2022-05-06T12:00:00Z,1
2022-05-06T12:00:01Z,null
2022-05-06T12:00:02Z,1
```

the part of the code which filters the "lonely" points to show is here below, but we'll need to reproduce the issue to see if the bug is here or elsewhere.

https://github.com/grafana/grafana/blob/99eaa0fc206a5829fee973c02e92793d8e0a3c2d/packages/grafana-ui/src/components/TimeSeries/utils.ts#L213-L248

**Comment by justinsmalley** [comment]
Thanks for looking at this. Below is the data set I used in my example (2760 data points in length) and I was able to reproduce it with the gdev-testdata datasource. I didn't see the effect with your data set, but when I pasted it 10 times in a row I was able to see it. I'm going to guess that it related to having more points than the panel has pixels wide?

I also found a very apparent way to see the effect is to drag your browser wider and narrower left to right and points  flicker in and out of visibility.

`,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,60,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,61,,,,,,60,,,,,,,,,,,,,,,,,,61,,,,,,60,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,61,,,,,,60,,,,,,59,,,,,,60,,,,,,,59,,,,,,,,,,,,,,,,,,,,,,,,58,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,57,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,56,,,,,,57,,,,,,56,,,,,,57,,,,,,,,,,,,56,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,55,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,56,,,,,55,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,54,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,53,,,,,54,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,55,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,56,,,,,,,,,,,,,,,,,,55,,,,,,56,,,,,,,,,,,,57,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,56,,,,,,57,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,56,,,,,,,,,,,,57,,,,,,,58,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,57,,,,,,,,,,,,,,,,,,,58,,,,,,59,,,,,,58,,,,,,,,,,,,,,,,,,,,,,,,57,,,,,,58,,,,,,,,,,,,,,,,,,,59,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,60,,,,,,,,,,,,61,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,62,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,61,,,,,,,,,,,,,,,,,,60,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,59,,,,,,60,,,,,,59,,,,,,60,,,,,,,,,,,,,,,,,,,,,,,,,59,,,,,,,,,,,,,,,,,60,,,,,,,,,,,,,59,,,,,,60,,,,,,,,,,,,59,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,58,,,,,,59,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,58,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,57,,,,,,58,,,,,,57,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,56,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,55,,,,,,56,,,,,,55,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,54,,,,,,53,,,,,,,,,,,,,,,,,,,52,,,,,,,,,,,51,,,,,,,,,,,,,,,,,,,,,,,,,52,,,,,,,54,,,,,,55,,,,,,56,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,57,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,58,,,,,,,,,,,,,,,,,,59,,,,,,,,,,,,,,,,,,60,,,,,,,,,,,,61,,,,,,,,,,,,,,,,,,,,,,,,60,,,,,,61,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,62,,,,,,61,,,,,,62,,,,,,61,,,,,,62,,,,,,,,,,,,,,,,,,,63,,,,,,,,,,,,,,,,,,,,,,,,64,,,,,,63,,,,,,64,,,,,,63,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,62,,,,,,,,,,,,,,,,,,,61,,,,,,,,,,,,62,,,,,,,,,,,,,,,,,,,,,,,,,63,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,64,,,,,,,,,,,,63,,,,,,,,,,,,,62,,,,,,,,,,,,,,,,,,,,`
